src/test/clojure/clojure/tools/gitlibs/impl_test.cljc

```diff
@@ -32,3 +40,267 @@
 
     ;; git - unknown transport with url rewrite in gitconfig (unusual, but do something useful)
     "work:repo.git" "ssh/work/repo"))
+
+;; Test helpers for concurrent clone coordination
+
+(defn- delete-dir-recursive
+  "Recursively deletes a directory and all its contents."
+  [^DirectoryInfo dir]
+  (when (.Exists dir)
+    (Directory/Delete (.FullName dir) true)))
+
+(defn- delete-file-if-exists
+  "Deletes a file if it exists."
+  [^String path]
+  (when (File/Exists path)
+    (File/Delete path)))
+
+(defn- deref-timely
+  "Dereferences a future with a timeout. Throws if timed out."
+  [fut timeout-ms timeout-msg]
+  (let [result (deref fut timeout-ms ::timeout)]
+    (when (= result ::timeout)
+      (throw (Exception. timeout-msg)))
+    result))
+
+(defn- create-temp-gitlibs-dir
+  "Creates a temporary gitlibs directory for test isolation."
+  []
+  (let [temp-dir (Path/Join (Path/GetTempPath) (str "gitlibs-test-" (DateTime/Now.Ticks)))]
+    (Directory/CreateDirectory temp-dir)
+    temp-dir))
+
+(defn- get-lockfile
+  "Returns the lockfile path for a given URL in the specified gitlibs dir."
+  [gitlibs-dir url]
+  (let [git-dir-file (cio/dir-info gitlibs-dir "_repos" (#'impl/clean-url url))]
+    (#'impl/lockfile-for-git-dir git-dir-file)))
+
+(defn- get-config-file-path
+  "Returns the config file path for a given URL in the specified gitlibs dir."
+  [gitlibs-dir url]
+  (let [git-dir-file (cio/dir-info gitlibs-dir "_repos" (#'impl/clean-url url))
+        config-file (cio/file-info git-dir-file "config")]
+    (.FullName config-file)))
+
+(defn- get-git-dir-path
+  "Returns the git directory path for a given URL in the specified gitlibs dir."
+  [gitlibs-dir url]
+  (.FullName (cio/dir-info gitlibs-dir "_repos" (#'impl/clean-url url))))
+
+;; Git wrapper process management
+
+(defn- create-git-wrapper-script
+  "Creates a git wrapper shell script that calls our CLR git wrapper function."
+  [scenario-name]
+  (let [script-path (Path/Join (Path/GetTempPath) (str "git-wrapper-" scenario-name ".sh"))
+        wrapper-fn (str "clojure.tools.gitlibs.impl/" scenario-name "-git-wrapper")]
+    (File/WriteAllText script-path
+      (str "#!/bin/bash\n"
+           "cljr -M:test -e \"(require 'clojure.tools.gitlibs.impl) ("
+           wrapper-fn " \\\"$@\\\")\"\n"))
+    ;; Make executable (on Unix-like systems)
+    script-path))
+
+;; Scenario 1a: We won the race, delete lockfile when git dir deleted
+(deftest test-scenario-1a-won-race-deleted-gitdir
+  (testing "Won race: delete lockfile when git dir deleted causes waiter to throw"
+    (let [test-url "https://github.com/clojure/spec.alpha.git"
+          temp-gitlibs (create-temp-gitlibs-dir)
+          _ (Environment/SetEnvironmentVariable "GITLIBS" temp-gitlibs)
```

Comment:
Don't set env vars in the test, we will send them as a map in ensure-git-dir's new arity.